### PR TITLE
chore: refactor Buffer usage in GitTree.js

### PR DESCRIPTION
**Pull request by addaleax**
Fix misleading/unnecessarily complicated usage of `Buffer`, namely
using `.from()` with an options object (which is ignored) or manually
converting from hex.

## I'm fixing a bug or typo

- [x] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README
- [x] squash merge the PR with commit message "fix: [Description of fix]"

### CODE REVIEW

**Comment by karma-pr-reporter** [comment]
## Test Results for bea24ec85bea99a05e85d361a459f08256c00961:
| Browser | Passed | Skipped | Failed | Time | Disconnected | Error |
|---------|--------|---------|--------|------|--------------|-------|
| HeadlessChrome 0.0.0 (Linux 0.0.0) | 254 | 1 | 0 | 1 min 25.834 secs | false | undefined |
| Firefox 65.0.0 (Ubuntu 0.0.0) | 252 | 1 | 2 | 2 mins 40.265 secs | false | undefined |
| Safari 11.1.2 (Mac OS X 10.13.6) | 254 | 1 | 0 | 3 mins 14.165 secs | false | undefined |
| Chrome Mobile 67.0.3396 (Android 7.1.1) | 254 | 1 | 0 | 4 mins 2.61 secs | false | undefined |
| Mobile Safari 11.0.0 (iOS 11.2.0) | 254 | 1 | 0 | 4 mins 16.717 secs | false | undefined |
| Edge 17.17134.0 (Windows 10 0.0.0) | 254 | 1 | 0 | 5 mins 23.814 secs | false | undefined |

File: src/models/GitTree.js [hunk 1]
```diff
@@ -108,9 +108,9 @@ export class GitTree {
       this._entries.map(entry => {
         let mode = Buffer.from(entry.mode.replace(/^0/, ''))
         let space = Buffer.from(' ')
-        let path = Buffer.from(entry.path, { encoding: 'utf8' })
+        let path = Buffer.from(entry.path, 'utf8')
         let nullchar = Buffer.from([0])
-        let oid = Buffer.from(entry.oid.match(/../g).map(n => parseInt(n, 16)))
+        let oid = Buffer.from(entry.oid, 'hex')
```
Comment (wmhilton) [inline src/models/GitTree.js]: lol! what was I thinking? 🤣

**Comment by wmhilton** [comment]
@addaleax Thank you so much!

**Comment by wmhilton** [comment]
> I noticed some potential linting errors, so I auto-formatted the code using prettier-standard. I hope you don't mind. 🤖

No... no you didn't? Bad bot. 😕 I need to add better branch permission checks or something.

**Pull request merged**